### PR TITLE
py-utils unittest fix's

### PR DESCRIPTION
**Pull request by veerendra-simha-garikipati**
Signed-off-by: Veerendra Garikipati <veerendra.garikipati@seagate.com>

# Problem Statement
- Fixed py-utils failing unittests mentioned in EOS-25091

# Design
-  https://jts.seagate.com/browse/EOS-25091

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: NA
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: NA
- [x] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Is there a change in filename/package/module or signature [Y/N]: Y
- [x] If yes for above point, Is a notification sent to all other cortx components [Y/N]: NA
- [x] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/test/support_bundle/test_support_bundle_cli.py [hunk 1]
```diff
@@ -269,7 +262,27 @@ def test_012_cli_verify_SB_get_status_after_cluster_stop(self):
         self.assertEqual(rc, 0)
         time.sleep(5)
 
-    def test_013_cli_verify_SB_generate_all_component(self):
+    def test_013_cortxcli_generate_status(self):
+        cmd = "cortxcli support_bundle generate 'sample comment' -c 'csm'"
```
Comment (rohit-k-dwivedi) [inline py-utils/test/support_bundle/test_support_bundle_cli.py:266]: please add description of test in a doc string

File: py-utils/test/support_bundle/test_support_bundle_cli.py [hunk 2]
```diff
@@ -269,7 +262,27 @@ def test_012_cli_verify_SB_get_status_after_cluster_stop(self):
...
+        self.assertEqual(rc, 0)
+        bundle_id = stdout.split('|')[1].strip()
+        time.sleep(15)
+        cmd = f"cortxcli support_bundle status '{bundle_id}'"
+        cmd_proc = SimpleProcess(cmd)
+        stdout, _, rc = cmd_proc.run()
+        self.assertIn('Success', stdout.decode('utf-8'))
+
+    def test_014_cortxcli_wrong_generate(self):
+        cmd = "cortxcli support_bundle generate 'sample comment' -c 'csmm'"
```
Comment (rohit-k-dwivedi) [inline py-utils/test/support_bundle/test_support_bundle_cli.py:280]: please add description of test in a doc string

File: py-utils/test/support_bundle/test_support_bundle_cli.py (hunk 1 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/support_bundle/test_support_bundle_cli.py:267]: Done

File: py-utils/test/support_bundle/test_support_bundle_cli.py (hunk 2 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/support_bundle/test_support_bundle_cli.py:282]: Done

File: py-utils/test/validator/test_path_validator.py [hunk 3]
```diff
@@ -18,10 +18,11 @@
 
 import unittest
 from cortx.utils.validator.v_path import PathV
+from cortx.utils.validator.error import VError
 
 class TestPathValidator(unittest.TestCase):
 	"""Test Path related validations."""
-	paths = ["dir:/", "file:/etc/hosts", "device:/dev/loop9"]
+	paths = ["dir:/", "file:/etc/hosts"]
```
Comment (vastradparayya) [inline py-utils/test/validator/test_path_validator.py:25]: This should be single quotes

File: py-utils/test/validator/test_path_validator.py [hunk 4]
```diff
@@ -18,10 +18,11 @@
...
 	host = "localhost"
```
Comment (vastradparayya) [inline py-utils/test/validator/test_path_validator.py:26]: This should be single quotes

File: py-utils/test/validator/test_path_validator.py [hunk 5]
```diff
@@ -34,5 +35,11 @@ def test_remote_path(self):
 
 		PathV().validate('exists', self.paths, self.host)
 
+	def test_path_neg(self):
+		"""Check negative paths."""
+		neg_path = ["device:/dev/loop9"]
```
Comment (vastradparayya) [inline py-utils/test/validator/test_path_validator.py:40]: This should be single quotes

**Review by vastradparayya — CHANGES_REQUESTED** [review]
Add single quotes wherever apply.

**Review by suryakumar1024 — APPROVED** [review]

File: py-utils/test/support_bundle/test_support_bundle.py [hunk 6]
```diff
@@ -36,7 +36,8 @@ def setUpClass(cls):
         """Test Setup class."""
         from cortx.utils.log import Log
         Log.init('support_bundle', '/var/log/cortx/utils/suppoort/', \
-            level='DEBUG', backup_count=5, file_size_in_mb=5)
+            level='DEBUG', backup_count=5, file_size_in_mb=5, \
+                syslog_server='localhost', syslog_port=514)
```
Comment (selvakumaar5496) [inline py-utils/test/support_bundle/test_support_bundle.py:40]: reduce 4 spaces. You are not going to next block or next line. 
```
Log.init('support_bundle', '/var/log/cortx/utils/suppoort/', \
    level='DEBUG', backup_count=5, file_size_in_mb=5, \
    syslog_server='localhost', syslog_port=514)
```

File: py-utils/test/support_bundle/test_support_bundle.py [hunk 7]
```diff
@@ -95,17 +96,6 @@ def test_004_verify_SB_get_status_success_multi_comp(self):
             self.assertEqual(status['status'][0]['result'], 'Success')
             self.assertEqual(status['status'][0]['result'], 'Success')
 
-    def test_005_verify_SB_get_status_empty_invalid_comp(self):
```
Comment (selvakumaar5496) [inline py-utils/test/support_bundle/test_support_bundle.py:98]: Please clarify why this test case is removed

File: py-utils/test/validator/test_path_validator.py (hunk 3 above)
Comment (selvakumaar5496) [inline py-utils/test/validator/test_path_validator.py:25]: can be changed to single quotes

File: py-utils/test/validator/test_pkg_validator.py [hunk 8]
```diff
@@ -18,10 +18,11 @@
 
 import unittest
 from cortx.utils.validator.v_pkg import PkgV
+from cortx.utils.validator.error import VError
 
 class TestRpmValidator(unittest.TestCase):
 	"""Test rpm related validations."""
-	pkg = ["lvm2-2.02.186-7.el7", "openldap-server"]
+	pkg = ["openldap-server"]
```
Comment (selvakumaar5496) [inline py-utils/test/validator/test_pkg_validator.py:25]: can be changed to single quotes

File: py-utils/test/validator/test_pkg_validator.py [hunk 9]
```diff
@@ -49,5 +50,11 @@ def test_remote_pip3_installed(self):
 		pkg = ["toml"]
 		PkgV().validate('pip3s', pkg, self.host)
 
+	def test_neg_rpm_installed(self):
+		"""Check if neagtive rpm pkg installed."""
+		neg_pkg = ["lvm2-2.02.186-7.el7"]
```
Comment (selvakumaar5496) [inline py-utils/test/validator/test_pkg_validator.py:55]: can be changed to single quotes

File: py-utils/test/validator/test_service_validator.py [hunk 10]
```diff
@@ -18,10 +18,11 @@
 
 import unittest
 from cortx.utils.validator.v_service import ServiceV
+from cortx.utils.validator.error import VError
 
 class TestServiceValidator(unittest.TestCase):
 	"""Test service related validations."""
-	services = ["rabbitmq-server", "sshd"]
+	services = ["sshd"]
```
Comment (selvakumaar5496) [inline py-utils/test/validator/test_service_validator.py:25]: can be changed to single quotes

File: py-utils/test/validator/test_service_validator.py [hunk 11]
```diff
@@ -34,5 +35,11 @@ def test_remote_service_running(self):
 
 		ServiceV().validate('isrunning', self.services, self.host)
 
+	def test_neg_service_running(self):
+		"""Check if ned services are running."""
+		neg_service = ["rabbitmq-server"]
```
Comment (selvakumaar5496) [inline py-utils/test/validator/test_service_validator.py:40]: can be changed to single quotes

**Review by selvakumaar5496 — CHANGES_REQUESTED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/test/support_bundle/test_support_bundle.py (hunk 7 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/support_bundle/test_support_bundle.py:98]: This test was added earlier like before fixing bug EOS-24882, where get_status is returning empty, but after fixing the bug this test is obsolete , so removed it

File: py-utils/test/support_bundle/test_support_bundle.py (hunk 6 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/support_bundle/test_support_bundle.py:40]: Done

File: py-utils/test/validator/test_path_validator.py (hunk 4 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_path_validator.py]: Done

File: py-utils/test/validator/test_path_validator.py (hunk 5 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_path_validator.py]: Done

File: py-utils/test/validator/test_path_validator.py (hunk 3 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_path_validator.py]: Done

File: py-utils/test/validator/test_pkg_validator.py (hunk 8 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_pkg_validator.py]: Done

File: py-utils/test/validator/test_pkg_validator.py (hunk 9 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_pkg_validator.py]: Done

File: py-utils/test/validator/test_service_validator.py (hunk 10 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_service_validator.py]: Done

File: py-utils/test/validator/test_service_validator.py (hunk 11 above)
Comment (veerendra-simha-garikipati) [reply py-utils/test/validator/test_service_validator.py]: Done

**Review by vastradparayya — APPROVED** [review]

**Review by rohit-k-dwivedi — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]

**Pull request merged**